gradle/init.gradle.kts

```diff
@@ -1,32 +1,46 @@
-fun resolvePluginPath(rootPath: java.nio.file.Path): java.nio.file.Path? {
-    if (rootPath.resolve("src").toFile().exists()) {
-        return null
-    }
-    if (rootPath.resolve("demo").toFile().exists()) {
-        return null
-    }
-    var path = rootPath
-    for (n in 1..4) {
-        val settingsFile = path.resolve("demo")
-        if (settingsFile.toFile().exists()) {
-            return path.toAbsolutePath()
-            break
-        } else {
-            path = path.resolve("..")
-            println("new path: ${path.toAbsolutePath()}")
-        }
-    }
-    return null
-}
-
 gradle.settingsEvaluated {
-    val property = "org.jsonschema2dataclass.local"
-    if (extra.has(property) && extra[property].toString().toBoolean()) {
-        println("root dir: ${getRootDir()}")
-        val path = resolvePluginPath(getRootDir().toPath().toAbsolutePath())
-        if (path != null) {
-            println("Include build:\"$path\"")
-            includeBuild(path)
+    val localPublishExtra = "org.jsonschema2dataclass.internal.local-publish"
+    val localVersionExtra = "org.jsonschema2dataclass.internal.git-version"
+    if (extra.has(localPublishExtra) && extra[localPublishExtra].toString().toBoolean()) {
+        val localVersion = extra[localVersionExtra]!!.toString()
+        pluginManagement {
+            repositories {
+                mavenLocal().content {
+                    includeGroupByRegex(".*.jsonschema2dataclass.*")
+                }
+                mavenCentral().content() {
+                    excludeGroupByRegex(".*jsonschema2dataclass.*")
+                }
+                gradlePluginPortal().content() {
+                    excludeGroupByRegex(".*jsonschema2dataclass.*")
+                }
+            }
+            resolutionStrategy {
+                eachPlugin {
+                    if (requested.id.name == "jsonschema2dataclass" && requested.version != localVersion) {
+                        useModule("org.jsonschema2dataclass:plugin:$localVersion")
+                    }
+                }
+            }
+        }
+        rootProject {
+            buildscript {
+                repositories {
+                    mavenLocal().content {
+                        includeGroupByRegex(".*.jsonschema2dataclass.*")
+                    }
+                    mavenCentral().content() {
+                        excludeGroupByRegex(".*jsonschema2dataclass.*")
+                    }
```

Comment:
have a look at `exclusiveContent`

Reply:
it's quite useful, thanks

Reply:
careful, I think you missed `forRepository`.